condor_log_Integer.py

```diff
@@ -0,0 +1,655 @@
+
+    return condor_log(
+        event_logs=args.files, conditions=args.where, attributes=args.groupby
+    )
+
+
+def condor_log(event_logs=None, conditions=None, attributes=None):
+
+    if event_logs is not None:
+        for f in event_logs:
+
+            jel = htcondor.JobEventLog(f)
+            job_time_slot = {}
+            job_event = []
+            returned_job = {}
+
+            new_attributes = {}
+            new_conditions = {}
+            false_type_in = []
+            false_type_in_attri = []
+            projected_job = []
+            no_condition = False
+            no_attributes = False
+
+            isIncluded = {}
+
+            if conditions is not None:
+                new_conditions, false_type_in = format_change.condition_format(
+                    conditions, false_type_in
+                )
+
+            if conditions is None:
+                no_condition = True
+
+            if attributes is None:
+                no_attributes = True
+
+            for event in jel.events(0):
+                job_status = event.type
+
+                if str(job_status) not in job_event:
+                    job_event.append(str(job_status))
+
+                job_c_id = str(event.cluster) + "." + str(event.proc)
+                if job_c_id not in isIncluded.keys():
+                    isIncluded[job_c_id] = True
+
+                info_dic = {}
+
+                if str(job_status) == JobStatus.JOB_INFO.value:
+                    info_dic = info.getmoreinfo(event)
+
+                    if no_condition is False:
+                        projected_job, false_type_in = info.project(
+                            event,
+                            new_conditions,
+                            info_dic,
+                            projected_job,
+                            false_type_in,
+                            isIncluded,
+                        )
+                    else:
+                        if job_c_id not in projected_job:
+                            projected_job.append(job_c_id)
+
+                    if no_attributes is False:
+                        returned_job, false_type_in_attri = info.groupby(
+                            event,
+                            attributes,
+                            projected_job,
+                            info_dic,
+                            returned_job,
+                            false_type_in_attri,
+                        )
+
+                cluster_id = event.cluster
+                if cluster_id not in job_time_slot:
+                    job_time_slot[cluster_id] = {}
+
+                job_id = event.proc
+                if job_id not in job_time_slot[cluster_id].keys():
+                    job_time_slot[cluster_id][job_id] = {}
+
+                timestamp = event.timestamp
+                dt_object = datetime.fromtimestamp(timestamp)
+                job_time_slot[cluster_id][job_id][str(job_status)] = dt_object
```

Comment:
I still really want to see a `Job` class that can capture this data in an object.